# HG changeset patch
# User Rik <rik@octave.org>
# Date 1478635788 28800
#      Tue Nov 08 12:09:48 2016 -0800
# Branch stable
# Node ID 4678729e8ae175973a9436890b3a1c3332e969a0
# Parent  4560273d100ce89d937ae3af9eb948d2386f3faf
Don't re-parse file in editor if in middle of debug session (bug #49171).

* file-editor-tab.cc (file_editor_tab::save_file): Check Fisdebugmode() before
calling exit_debug_and_clear().

diff --git a/libgui/src/m-editor/file-editor-tab.cc b/libgui/src/m-editor/file-editor-tab.cc
--- a/libgui/src/m-editor/file-editor-tab.cc
+++ b/libgui/src/m-editor/file-editor-tab.cc
@@ -1835,17 +1835,18 @@ file_editor_tab::save_file (const QStrin
 
   // get the absolute path (if existing)
   QFileInfo file_info = QFileInfo (saveFileName);
   QString file_to_save;
   if (file_info.exists ())
     {
       file_to_save = file_info.canonicalFilePath ();
       // Force reparse of this function next time it is used (bug #46632)
-      if (!exit_debug_and_clear (file_to_save, file_info.baseName ()))
+      if ((Fisdebugmode ())(0).is_true ()
+          && ! exit_debug_and_clear (file_to_save, file_info.baseName ()))
         return;
     }
   else
     file_to_save = saveFileName;
   QFile file (file_to_save);
 
   // stop watching file
   QStringList trackedFiles = _file_system_watcher.files ();
